typing_inspect.py

```diff
@@ -322,6 +386,36 @@ def get_args(tp, evaluate=None):
     return ()
 
 
+def get_bound(tp):
+    """Returns the type bound to a `TypeVar` if any. It the type is not a `TypeVar`, a `TypeError` is raised
+
+    Examples::
+
+        get_bound(TypeVar('T')) == None
+        get_bound(TypeVar('T', bound=int)) == int
+    """
+
+    if is_typevar(tp):
+        return getattr(tp, '__bound__', None)
+    else:
+        raise TypeError("type is not a `TypeVar`: " + str(tp))
+
+
+def get_constraints(tp):
+    """Returns the constraints of a `TypeVar` if any. It the type is not a `TypeVar`, a `TypeError` is raised
+
+    Examples::
+
+        get_constraints(TypeVar('T')) == ()
+        get_constraints(TypeVar('T', int, str)) == (int, str)
+    """
+
+    if is_typevar(tp):
+        return getattr(tp, '__constraints__', ())
+    else:
+        raise TypeError("type is not a `TypeVar`: " + str(tp))
+
+
```

Comment:
These are nice additions. One can combine them with the current version of `is_optional_type()` to get exactly the behaviour one needs.

Reply:
ok